plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/typehints/CheckEdgeTargetResultAction.java

```diff
@@ -0,0 +1,79 @@
+/********************************************************************************
+ * Copyright (c) 2023 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.features.typehints;
+
+import java.util.Optional;
+
+import org.eclipse.glsp.server.actions.ResponseAction;
+
+/**
+ * Response Action for a {@link RequestCheckEdgeAction}. It provides
+ * a boolean indicating whether the requested element is a valid target
```

Comment:
the `requested element` is a bit confusing. I added some comments in the client change as well, I think the doc could be a bit clearer in what this action does and avoid the word `target` in the context of edges.